packages/expo-modules-test-core/src/mockgen.ts

```diff
@@ -0,0 +1,164 @@
+    case ts.SyntaxKind.BooleanKeyword:
+      return [ts.factory.createReturnStatement(ts.factory.createFalse())];
+    case ts.SyntaxKind.NumberKeyword:
+      return [ts.factory.createReturnStatement(ts.factory.createNumericLiteral('0'))];
+    case ts.SyntaxKind.VoidKeyword:
+      return [];
+    case ts.SyntaxKind.ArrayType:
+      return [ts.factory.createReturnStatement(ts.factory.createArrayLiteralExpression())];
+  }
+  return [];
+}
+
+function wrapWithAsync(tsType: ts.TypeNode) {
+  return ts.factory.createTypeReferenceNode('Promise', [tsType]);
+}
+
+function getMockedFunctions(functions: Closure[], async = false) {
+  return functions.map((fnStructure) => {
+    const name = ts.factory.createIdentifier(fnStructure.name);
+    const returnType = mapSwiftTypeToTsType(fnStructure.types?.returnType);
+    const func = ts.factory.createFunctionDeclaration(
+      [
+        ts.factory.createToken(ts.SyntaxKind.ExportKeyword),
+        async ? ts.factory.createToken(ts.SyntaxKind.AsyncKeyword) : undefined,
+      ].filter((f) => !!f) as ts.ModifierToken<any>[],
+      undefined,
+      name,
+      undefined,
+      fnStructure?.types?.parameters.map((p) =>
+        ts.factory.createParameterDeclaration(
+          undefined,
+          undefined,
+          p.name,
+          undefined,
+          mapSwiftTypeToTsType(p.typename),
+          undefined
+        )
+      ) ?? [],
+      async ? wrapWithAsync(returnType) : returnType,
+      ts.factory.createBlock(getMockReturnStatements(returnType), true)
+    );
+    return func;
+  });
+}
+
+function getTypesToMock(module: OutputModuleDefinition) {
+  const foundTypes: string[] = [];
+
+  Object.values(module)
+    .flatMap((t) => (Array.isArray(t) ? t?.map((t2) => (t2 as Closure)?.types) : [] ?? []))
+    .forEach((types: ClosureTypes | null) => {
+      types?.parameters.forEach(({ typename }) => {
+        foundTypes.push(maybeUnwrapSwiftArray(typename));
+      });
+      types?.returnType && foundTypes.push(maybeUnwrapSwiftArray(types.returnType));
+    });
+  return new Set(
+    foundTypes.filter((ft) => mapSwiftTypeToTsType(ft).kind === ts.SyntaxKind.TypeReference)
+  );
+}
+
+function getMockedTypes(types: Set<string>) {
+  return Array.from(types).map((type) => {
+    const name = ts.factory.createIdentifier(type);
+    const typeAlias = ts.factory.createTypeAliasDeclaration(
+      [ts.factory.createToken(ts.SyntaxKind.ExportKeyword)],
+      name,
+      undefined,
+      ts.factory.createKeywordTypeNode(ts.SyntaxKind.AnyKeyword)
+    );
+    return typeAlias;
+  });
+}
+
+function getMockForModule(module: OutputModuleDefinition) {
+  return ([] as (ts.TypeAliasDeclaration | ts.FunctionDeclaration)[]).concat(
+    getMockedTypes(getTypesToMock(module)),
+    getMockedFunctions(module.functions),
+    getMockedFunctions(module.asyncFunctions, true)
+  );
+}
+
+export function printModules(modules: OutputModuleDefinition[]) {
```

Comment:
Name of the function is a bit confusing. I know that TS uses a "printer", but it's confusing if you don't know that. Maybe just `generateMocks`?